packages/babel-types/src/builders/generated/index.ts

```diff
@@ -45,179 +57,374 @@ export function binaryExpression(
     | "<=",
   left: t.Expression | t.PrivateName,
   right: t.Expression,
-): t.BinaryExpression {
-  return builder("BinaryExpression", ...arguments);
-}
-export function interpreterDirective(value: string): t.InterpreterDirective {
-  return builder("InterpreterDirective", ...arguments);
-}
-export function directive(value: t.DirectiveLiteral): t.Directive {
-  return builder("Directive", ...arguments);
-}
-export function directiveLiteral(value: string): t.DirectiveLiteral {
-  return builder("DirectiveLiteral", ...arguments);
+) {
+  const node = {
+    type: "BinaryExpression",
+    operator: operator,
+    left: left,
+    right: right,
+  } as t.BinaryExpression;
+  validateNode(node);
+  return node;
+}
+export function interpreterDirective(value: string) {
+  const node = {
+    type: "InterpreterDirective",
+    value: value,
+  } as t.InterpreterDirective;
+  validateNode(node);
+  return node;
+}
+export function directive(value: t.DirectiveLiteral) {
+  const node = {
+    type: "Directive",
+    value: value,
+  } as t.Directive;
+  validateNode(node);
+  return node;
+}
+export function directiveLiteral(value: string) {
+  const node = {
+    type: "DirectiveLiteral",
+    value: value,
+  } as t.DirectiveLiteral;
+  validateNode(node);
+  return node;
 }
 export function blockStatement(
   body: Array<t.Statement>,
   directives?: Array<t.Directive>,
-): t.BlockStatement {
-  return builder("BlockStatement", ...arguments);
-}
-export function breakStatement(label?: t.Identifier | null): t.BreakStatement {
-  return builder("BreakStatement", ...arguments);
+) {
+  const node = {
+    type: "BlockStatement",
+    body: body,
+    directives: directives === undefined ? [] : directives,
+  } as t.BlockStatement;
+  validateNode(node);
+  return node;
+}
+export function breakStatement(label?: t.Identifier | null) {
+  const node = {
+    type: "BreakStatement",
+    label: label === undefined ? null : label,
+  } as t.BreakStatement;
+  validateNode(node);
+  return node;
 }
 export function callExpression(
   callee: t.Expression | t.V8IntrinsicIdentifier,
   _arguments: Array<
     t.Expression | t.SpreadElement | t.JSXNamespacedName | t.ArgumentPlaceholder
   >,
-): t.CallExpression {
-  return builder("CallExpression", ...arguments);
+) {
+  const node = {
+    type: "CallExpression",
+    callee: callee,
+    arguments: _arguments,
+    optional: null,
+    typeArguments: null,
+    typeParameters: null,
+  } as t.CallExpression;
+  validateNode(node);
+  return node;
 }
 export function catchClause(
   param: t.Identifier | t.ArrayPattern | t.ObjectPattern | null | undefined,
   body: t.BlockStatement,
-): t.CatchClause {
-  return builder("CatchClause", ...arguments);
+) {
+  const node = {
+    type: "CatchClause",
+    param: param === undefined ? null : param,
```

Comment:
Can we replace `# === undefined ? a : b` by default parameters, if it is one of the builder keys.

Reply:
yeah, actually already tried that - let me push it